torch/csrc/jit/profiling_graph_executor_impl.cpp

```diff
@@ -112,7 +112,7 @@ ExecutionPlan ProfilingGraphExecutorImpl::getPlanFor(Stack& stack) {
   // TODO: insert grad propagation
   bool needs_gradient = getProfilingMode()
       ? needsGradientInProfilingMode(copy->block())
-      : needsGradient(copy);
+      : false;
```

Comment:
This seems wrong. I would not expect this to change with this patch.

Reply:
could you please elaborate why this is wrong. this turns a profiling executor into a simple as per our discussion if `getProfilingMode() == false`